storage/remote/queue_manager.go

```diff
@@ -887,14 +884,15 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 		return err
 	}
 
-	span.SetTag("samples", len(samples))
-	span.SetTag("requestSize", len(req))
-
 	retries := 0
-	// Anonymous function is required so that retries value is not captured as zero.
-	defer func() { span.SetTag("retries", retries) }()
 
 	for {
+		span, ctx := opentracing.StartSpanFromContext(ctx, "Remote Send Batch")
+		span.SetTag("samples", len(samples))
+		span.SetTag("requestSize", len(req))
+		span.SetTag("retries", retries)
```

Comment:
Rather than `retries` could this be something like `try`? If I see a `retries` tag in a trace I would expect all of those retries to be part of that one trace.

Reply:
Agreed.